OpenRA.Game/Traits/World/ScreenMap.cs

```diff
@@ -33,6 +33,7 @@ public class ScreenMap : IWorldLoaded
 		readonly Cache<Player, SpatiallyPartitioned<FrozenActor>> partitionedFrozenActors;
 		readonly SpatiallyPartitioned<Actor> partitionedActors;
 		WorldRenderer worldRenderer;
+		public Viewport Viewport { get { return worldRenderer.Viewport; } }
```

Comment:
Requiring a WorldRenderer to access the Viewport (along with other render-only data) was a deliberate decision to force people to stop mixing simulation and rendering code is unmaintainable ways.  This makes the viewport accessible to anyone who has a world, so :-1: